File: src/main/java/com/procurement/contracting/model/dto/ContractStatusDetails.java
package com.procurement.contracting.model.dto;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;
import java.util.HashMap;
import java.util.Map;

public enum ContractStatusDetails {
    CONTRACT_PROJECT("contractProject"),
    ACTIVE("active"),
    VERIFIED("verified"),
    CANCELLED("cancelled"),
    COMPLETE("complete"),
    UNSUCCESSFUL("unsuccessful");

    private static final Map<String, ContractStatusDetails> CONSTANTS = new HashMap<>();

    static {
        for (final ContractStatusDetails c : values()) {
            CONSTANTS.put(c.value, c);
        }
    }

    private final String value;

    ContractStatusDetails(final String value) {
        this.value = value;
    }

    @JsonCreator
    public static ContractStatusDetails fromValue(final String value) {
        final ContractStatusDetails constant = CONSTANTS.get(value);
        if (constant == null) {
            throw new IllegalArgumentException(value);
        }
        return constant;
    }

    @Override
    public String toString() {
        return this.value;
    }

    @JsonValue
    public String value() {
        return this.value;
    }
}

File: src/main/kotlin/com/procurement/contracting/domain/model/contract/status/ContractStatusDetails.kt
package com.procurement.contracting.domain.model.contract.status

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonValue
import com.procurement.contracting.domain.model.EnumElementProvider

enum class ContractStatusDetails(@JsonValue override val key: String) : EnumElementProvider.Key {
    CONTRACT_PROJECT("contractProject"),
    CONTRACT_PREPARATION("contractPreparation"),
    APPROVED("approved"),
    SIGNED("signed"),
    VERIFICATION("verification"),
    VERIFIED("verified"),
    ISSUED("issued"),
    APPROVEMENT("approvement"),
    EXECUTION("execution"),
    EMPTY("empty");

    override fun toString(): String = key

    companion object : EnumElementProvider<ContractStatusDetails>(info = info()) {
        @JvmStatic
        @JsonCreator
        fun creator(name: String) = ContractStatusDetails.orThrow(name)
    }
}